src/app/loadout/fashion/FashionDrawer.tsx

```diff
@@ -0,0 +1,396 @@
+  );
+
+  const [modsByBucket, setModsByBucket] = useState(loadout.parameters?.modsByBucket ?? {});
+  const isShader = (h: number) =>
+    defs.InventoryItem.get(h)?.plug?.plugCategoryHash === PlugCategoryHashes.Shader;
+  const modHashes = Object.values(modsByBucket).flat();
+  const shaders = modHashes.filter(isShader);
+  const ornaments = modHashes.filter((h) => !isShader(h));
+
+  // TODO: if they add an armor piece that can't slot the selected mods (ornaments), clear them
+
+  const header = (
+    <>
+      <h1>{t('FashionDrawer.Title')}</h1>
+    </>
+  );
+
+  const footer = ({ onClose }: { onClose(): void }) => (
+    <>
+      <button
+        type="button"
+        className="dim-button"
+        onClick={() => {
+          onModsByBucketUpdated(modsByBucket);
+          onClose();
+        }}
+      >
+        {t('FashionDrawer.Accept')}
+      </button>
+    </>
+  );
+
+  const handlePlugSelected = ({
+    item,
+    socket,
+    plugHash,
+  }: {
+    item: DimItem;
+    socket: DimSocket;
+    plugHash: number;
+  }) => {
+    setModsByBucket((modsByBucket) => {
+      // Start by removing any mods that belong to the same socket as this
+      const existingMods = (modsByBucket[item.bucket.hash] ?? []).filter(
+        (mod) => !socket.plugSet?.plugs.some((p) => p.plugDef.hash === mod)
+      );
+
+      return {
+        ...modsByBucket,
+        // Add in the new mod
+        [item.bucket.hash]: [...existingMods, plugHash],
+      };
+    });
+  };
+
+  const handleUseEquipped = () => {
+    const newModsByBucket = _.mapValues(armorItemsByBucketHash, (item) => {
+      const cosmeticSockets = item.sockets
+        ? getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.ArmorCosmetics)
+        : [];
+      return _.compact(cosmeticSockets.map((s) => s.plugged?.plugDef.hash));
+    });
+
+    setModsByBucket(newModsByBucket);
+  };
+
+  const handleSyncShader = () => {
+    const shaders = Object.values(modsByBucket).flat().filter(isShader);
+
+    const groupedShaders = _.groupBy(shaders, (h) => h);
+    const mostCommonShaders = _.maxBy(Object.values(groupedShaders), (shaders) => shaders.length);
+    if (!mostCommonShaders) {
+      return;
+    }
+
+    setModsByBucket((modsByBucket) =>
+      Object.fromEntries(
+        LockableBucketHashes.map((bucketHash) => [
+          bucketHash,
+          [...(modsByBucket[bucketHash] ?? []).filter((h) => !isShader(h)), mostCommonShaders[0]],
+        ])
+      )
+    );
+  };
+
+  const handleSyncOrnament = () => {
+    const isShader = (h: number) =>
+      defs.InventoryItem.get(h)?.plug?.plugCategoryHash === PlugCategoryHashes.Shader;
+    const ornaments = Object.values(modsByBucket)
+      .flat()
+      .filter((h) => !isShader(h));
+
+    const groupedOrnaments = _.groupBy(ornaments, (h) => {
+      const collectibleHash = defs.InventoryItem.get(h)?.collectibleHash;
+      return collectibleHash && defs.Collectible.get(collectibleHash)?.parentNodeHashes[0];
+    });
```

Comment:
I am not across collections really but I am guessing this needs to also check if the inventory item has a `canInsert` value of true?